### ISSUE TITLE
Space bar scroll hides text behind the search bar

### ISSUE THREAD

**Issue by wwhyte-si** [TeamMentor/TM_4_0_Design#600]
When I scroll down using the space bar I would like the bottom line of the previous screen to scroll up to be immediately under the search bar -- i.e. so that I can see the last line of context from my previous view, followed by the next screenful of content.

However, the bottom line scrolls up further than that, it seems like it is going to the top of the search bar. This means that the search bar is hiding some of the results and I need to arrow-key back up a couple of lines.

The first image below shows the first results screen I see. The second image below shows what I see after one space bar push. The final image shows what I'd expect to see.

![first screen](https://cloud.githubusercontent.com/assets/4628742/6722180/05d070f4-cd93-11e4-865d-b67635cf98a7.png)
![second screen](https://cloud.githubusercontent.com/assets/4628742/6722184/0d14caae-cd93-11e4-8b56-04dedb654137.png)
![better screen](https://cloud.githubusercontent.com/assets/4628742/6722185/0fa209b2-cd93-11e4-94d5-9323c1b2c24d.png)

**Comment by roman87** [comment]
Thanks. Yes very good point. This will be fixed in the Flare (JS) version. For this version this will probably need to be a known limitation, at least for the time being. I will follow up with you when it is fixed.

**Comment by wwhyte-si** [comment]
Thanks!

- sent from my phone

**Comment by luchia** [comment]
Prioritising my Sprint 3 work.

**Comment by luchia** [comment]
I've done some researching today and it looks like this is dependant on the users browser. It's possible that we could adapt a JS library to fix this but in Jade version of TM it's impossible to do.

**Comment by luchia** [comment]
Changing back to P3.

**Comment by roman87** [comment]
Still showing up as P0, moving it to P1

**Comment by luchia** [comment]
This isn't possible to do without doing some weird hacks. It's browser standard, it takes you to one screen-height further down than you were before. I don't know if this is something we want to fix or not?

**Comment by wwhyte-si** [comment]
If the space bar takes you down one screen height we shouldn’t use widgets
that effectively change the definition of screen height without telling the
browser. If you’re used to using the space bar to page down, as I am, this
behavior is terribly frustrating.



Cheers,



William



*From:* Luchia Bloomfield [mailto:notifications@github.com]
*Sent:* Saturday, May 16, 2015 9:05 PM
*To:* TeamMentor/TM_4_0_Design
*Cc:* wwhyte-si
*Subject:* Re: [TM_4_0_Design] Space bar scroll hides text behind the
search bar (#600)



This isn't possible to do without doing some weird hacks. It's browser
standard, it takes you to one screen-height further down than you were
before. I don't know if this is something we want to fix or not?

—
Reply to this email directly or view it on GitHub
<https://github.com/TeamMentor/TM_4_0_Design/issues/600#issuecomment-102717650>
.

**Comment by wwhyte-si** [comment]
In other words, I would rather not have the search bar as a floating
element within the results screen, but as an element in a different screen.
Can we do that?

- sent from my phone

**Comment by roman87** [comment]
Yes, we should fix this issue, just not in this version :) This should be addressed in Flare, which is a jazzier version of TM.

**Comment by wwhyte-si** [comment]
I'm fine with fixing it later :-)

**Comment by luchia** [comment]
[JS library that will handle this](https://craig.is/killing/mice).

**Comment by luchia** [comment]
I ended up not having to use a JS library because I spent the time planning Flare properly (no CSS hacks). Here's what it looks like when you use the space bar to move down the page -

### Article

![article](https://cloud.githubusercontent.com/assets/4547737/8901611/f6c02f9c-3452-11e5-91fc-82401ca14a80.gif)

### Main

![main](https://cloud.githubusercontent.com/assets/4547737/8901557/8e593cdc-3452-11e5-9f0a-327904f77b9e.gif)

### Navigate

![navigate](https://cloud.githubusercontent.com/assets/4547737/8901577/aa71ecc0-3452-11e5-8618-dc073f3d8ebe.gif)

**Comment by romichg** [comment]
There is a small issue here. In order for me to use the space bar, I actually need to click on the page. For example, if I click on the article title, I then need to click on the page for the space bar to work.

**Comment by luchia** [comment]
Okay, we will need to do this with JS then. I sent Dinis a message on Slack regarding this. Here's a few more options to consider:

https://github.com/bfanger/angular-keyboard
http://chieffancypants.github.io/angular-hotkeys/

**Comment by romichg** [comment]
@DinisCruz any ideas how we can get this done?

**Comment by DinisCruz** [comment]
this is related to the pagination issues that @michaelhidalgo is working on (and scrolling) so pushing this to him
